mcs/class/System/System.Net.Sockets/Socket.cs

```diff
@@ -1250,37 +1314,52 @@ public bool ConnectAsync (SocketAsyncEventArgs e)
 			if (e.RemoteEndPoint == null)
 				throw new ArgumentNullException ("remoteEP");
 
-			e.curSocket = this;
-			e.Worker.Init (this, e, SocketOperation.Connect);
+			SocketAsyncResult sockares = new SocketAsyncResult (this, s => SocketAsyncEventArgsCallback (ConnectAsyncCallback, s), e, SocketOperation.Connect);
```

Comment:
Renaming the variable from result to sockares introduced confusion in the diff that makes it hard to see what's changing.